apps/api/src/resources/Event/resolver.ts

```diff
@@ -152,9 +152,33 @@ export class EventResolver {
       newResource.flierLink = url;
     }
 
+    let charPool: string = "0123456789abcdefghijklmnopqrstuvwxyz";
+    let keyPlaceholder: string = "";
+    for(let i = 0; i < 4; i++)
+      keyPlaceholder += charPool.charAt(Math.floor(Math.random() * 36));
+    newResource.urlKey = keyPlaceholder;
```

Comment:
you are not handling collisions, if your intention is to allow collisions (duplicate urlKey) then you need to handle the logic of what the client should see when a duplicate occurs